### ISSUE TITLE
Consider putting ffmpeg in /lib/<abi>/libffmpeg.so instead

### ISSUE THREAD

**Issue by mortyccp** [hiteshsondhi88/ffmpeg-android-java#131]
According to [ABI Management on the Android Platform](https://developer.android.com/ndk/guides/abis.html), android have automatic extraction of native code at install time. If we fake the ffmpeg as a shared library, we can reduce code to determine which cpu and copy ffmpeg by ourselves.

**Comment by hiteshsondhi88** [comment]
Yes thats a valid point. But then how we will be able to execute commands. 

To achieve above I think we need to create a custom class or edit ffmpeg.c class and add jni methods, which is actually more effort then what we are doing now.

**Comment by mortyccp** [comment]
I think you misunderstand what I suggest. We can still compile ffmpeg as command line tools but just pretend it as `.so` so the android build system will do the automatic extraction for us. I think way we can have simpler logic on loading the correct binary of ffmpeg.

Of course, the ideal way to do this would be what you say. But JNI is too complex for me.

Here is the sample code I am using right now.

```
String ffmpeg = new File(getApplicationInfo().nativeLibraryDir + File.separator + "libffmpeg.so").getAbsolutePath();
FFmpegExecuteAsyncTask ffmpegExecuteAsyncTask = new FFmpegExecuteAsyncTask(cmd5, timeout, new FFmpegExecuteResponseHandler() {});
ffmpegExecuteAsyncTask.execute();
```
